Newsletters: add 112 (2020-08-26)

harding opened the pull request:
- [x] Update releases/RCs on Tuesday @harding 
- [x] Selected Q&A from Bitcoin StackExchange @bitschmidty 
- [x] `Bitcoin Core #14582` and `#19743` wallet: always do partial spends @jonatack 
- [x] `BIPs #955` BIP174: add hash preimage fields to inputs @dongcarl

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,109 @@
+This week's newsletter links to a discussion about routed coinswaps and
+includes our regular sections with summaries of questions and answers
+from the Bitcoin StackExchange, releases and release candidates, and
+notable changes to popular Bitcoin infrastructure software
```
missing full stop here and line 30.

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,109 @@
+
+- [Bitcoin Core #19550][] adds a new `getindexinfo` RPC that indicates
+  which optional indexes have been enabled and how many blocks they've
+  indexed so far.
```
for each index, it indicates the sync status and blockheight

jonatack reviewed (commented):
Happy Sunday :) a couple of things on first read-through.

jnewbery commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,109 @@
+  implementation to the Bitcoin-Dev mailing list as a follow-up to his
+  previous post in May (see [Newsletter #100][news100 coinswap]).
+  Discussion from ZmnSCPxj and Nadav Kohen focused on ensuring the
+  protocol was safe both in its use of cryptography and in ensuring that
```
Remove repeated ensuring: "ensuring the protocol was safe in ... ensuring that ..."

jnewbery commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,109 @@
+  (because, if feerates do increase beyond what was predicted, the node
+  can fee bump its commitment transaction).  Various degrees of support
+  for anchor outputs have already been merged into several LN
+  implementations monitored by Optech.
```
perhaps remove the "monitored by Optech"

jonatack replied:
I hesitated to comment on this as well, wondering "which impls does Optech not monitor?" while thinking that, knowing Dave, there must be a specific reason for writing this.

harding replied:
We (I) monitor the ones listed in in the *notable code changes* section.  The reason for appending that text was as a lead-in for linking to our summaries of the PRs for anchor outputs from past newsletters; however, I got lazy and dropped that.  I'll drop this vestigial text too.

jnewbery commented:
Changes look good to me. Thanks @harding

adamjonas commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,110 @@
+
+- [Bitcoin Core #19550][] adds a new `getindexinfo` RPC that lists each
+  optional index that has been enabled, whether it is fully synced to
+  the tip of the block chain, and (in case it hasn't been fully synced)
```
I don't think parentheses quite fit here. The case of not being fully synced is equal to the fully synced case which is not in parentheses in the previous clause.

```suggestion
  the tip of the block chain, and if not fully synced,
```

adamjonas commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,110 @@
+  so that they can each take a `locktime` parameter that specifies the
+  nLockTime of the transaction to create.  The PR notes that this is
+  "required for [dual funding][bolts #524], where the opener sets [the
+  locktime]".
```
```suggestion
  locktime]."
```

harding replied:
What you propose is indeed the normal American style but a number of programmers have complained to me about "closing" the sentence before "closing" a quote that was "opened" within that sentence.  I did a quick search and it seems maybe other programing documentation writers have adopted the same policy: `https://github.com/MicrosoftDocs/microsoft-style-guide/issues/99`

So maybe this is something we discuss?  A quick grep says we already have many cases of `\."`, so I think this is ok to leave for now.  Does anyone have a strong opinion?  I have a slight preference towards the normal `."` style Jonas recommends.

jonatack replied:
+1 for `."`

adamjonas commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,110 @@
+  Child Pays For Parent ([CPFP][topic cpfp]) fee bumping.   With this
+  change, it becomes possible to fee bump commitment transactions that
+  may have been signed days or weeks earlier, when the current feerates
+  would've been hard to predict.  In practice, LN nodes using anchor
+  outputs should pay lower fees normally (because there's no longer any
+  incentive to overestimate fees) and also have greater security
+  (because, if feerates do increase beyond what was predicted, the node
+  can fee bump its commitment transaction).  Various degrees of support
```
Suggest breaking into two sentences to match the two benefits of anchor outputs:
```suggestion
  would've been hard to predict.  In practice, LN nodes using anchor
  outputs should normally pay lower fees because there's no longer any
  incentive to overestimate fees. Anchor outputs also provide greater security
  because, if feerates do increase beyond what was predicted, the node
  can fee bump its commitment transaction.  Various degrees of support
```

adamjonas commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,110 @@
+- [LND 0.11.0-beta][lnd 0.11.0-beta] is now released.   This new major
+  version allows accepting [large channels][topic large channels] (by
+  default, this is off) and contains numerous improvements to its
+  backend features that may be of interest to advanced users (see the
+  [release notes][lnd 0.11.0-beta]).
```
Probably fine just to link once:
```suggestion
  backend features that may be of interest to advanced users.
```

jonatack commented:
Sorry for the rebase :frowning:

bitschmidty commented:
Add stack exchange commit

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -40,7 +40,23 @@ answers posted since our last update.*
+  Michael Folkson explains the considerations involving a potential testnet
+  reset or introduction of a new [signet][topic signet] network. A [related Stack
+  Exchange question][stack exchange signet setup] this month also summarizes two
+  options for working with signet.
```
Probably out of scope, but a signet meta-issue was opened today by Michael as well and discussion has begun: https://github.com/bitcoin/bitcoin/issues/19787

jnewbery commented:
New sections by jon, mike and carl all lookd good to me.

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -111,7 +111,11 @@ release candidates.*
-- [BIPs #955][] BIP174: add hash preimage fields to inputs FIXME:dongcarl
+- [BIPs #955][] updates [BIP174][]to standardize supplying hash preimages in
+  [PSBT][topic psbt] input records. The standardization of these preimage fields
+  were [found to be necessary for miniscript aware finalizers][psbt ext], though
```
s/were/was/ and missing hyphen in "miniscript aware"

harding commented:
Pushed edits, thanks everyone for the feedback!  @jonatack I made some significant edits to your description, mostly just rephrasing things so they sounded a bit simpler to me.  Please let me know if you object and I'll revert.  Thanks for that nice description!

@bitschmidty section and @dongcarl PR LGTM, thanks!

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -101,9 +101,9 @@ release candidates.*
-  the tip of the block chain, and (in case it hasn't been fully synced)
-  how many blocks have been indexed so far.
+  optional index that has been enabled, how many blocks have been
+  indexed so far, and whether those blocks constitue syncing all the way
```
constitue?

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -120,9 +120,9 @@ release candidates.*
 - [BIPs #955][] updates [BIP174][] to standardize supplying hash preimages in
   [PSBT][topic psbt] input records. The standardization of these preimage fields
-  were [found to be necessary for miniscript aware finalizers][psbt ext], though
+  were [found to be necessary for miniscript-aware finalizers][psbt ext], though
```
I still think s/were/was/ here, as the subject is the (singular) standardization of the fields, not the (plural) fields.

harding replied:
Sorry, I looked right past that in your original feedback.

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -79,26 +79,33 @@ release candidates.*
+  to the same address when only a subset of those inputs might be needed.
+  For this reason, `avoidpartialspends` is disabled by default unless
+  the `avoid_reuse` flag is enabled for the wallet (see [Newsletter
+  #52][news52 avoid_reuse]).  The new `maxapsfee` feature now gives
```
The first paragraph is much better now.

The second paragraph perhaps lost some meaning and transition that was provided by "It is for this default case that `maxapsfee` was conceived", though I'll concede "conceived" may be a bit of an unusual word for the writing here. 

ISTM that "The new `maxapsfee` feature..." is an intro phrase that should be the start of a new paragraph.

I thought that "not more than" and "not greater than" seemed simpler and more concise than "less than or equal to" but deferring to your judgement.

jonatack replied:
Thanks for adding the topic links!

harding replied:
@jonatack tried to address your concerns.

The problem I had with "**are not**" was that it always appeared in the same clause as "**are avoided"**.  Having two negatives in the same clause tends to make it hard to parse, at least for a attention deficit like me.  E.g., I found this challenging:

> partial spends are avoided if the fees are not higher than regular coin selection

However, if my edits aren't an obvious improvement, they're not good enough, so I tried again.  Let me know if you think this is better or if I should go back to your original.

harding commented:
Note: given [recent comments](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-August/018135.html) by Antoine Riard on the coinswap thread, I removed the conclusion from the corresponding bullet point in the newsletter and replaced it with a "discussion is still ongoing" boilerplate.

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -90,22 +90,24 @@ release candidates.*
   the `avoid_reuse` flag is enabled for the wallet (see [Newsletter
-  #52][news52 avoid_reuse]).  The new `maxapsfee` feature now gives
-  users a choice between three configurations:
+  #52][news52 avoid_reuse]).  It is for this default case that maxapsfee
```
unsure if "maxapsfee" was intentionally left without code markdown in this line, but it's probably  fine like this as well

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -90,22 +90,24 @@ release candidates.*
-       is less than or equal to 1,000 sats.
+    3. `maxapsfee` set to greater than `0`: partial spend avoidance is
+       used whenever the maximum additional cost it adds to the
+       transaction is the passed amount.  For example,
```
"whenever the maximum additional cost it adds to the transaction is the passed amount" seems less clear to me than the first and second versions, but I'll defer to any other opinion here

jonatack reviewed (commented):
Seems better, the formatting too.

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+  previous post in May (see [Newsletter #100][news100 coinswap]).
+  Comments focused on checking that the
+  protocol was safe both in its use of cryptography and in ensuring that
+  the expected transactions would get confirmed (rather than alternative
```
nit, s/get/be/

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+  flexibility in their setups and developers maintainability and security benefits.
+
+- [What's the most efficient way to create a raw transaction with a specific fee rate?]({{bse}}98392)
+  User Darius asks about the interplay between UTXO selection and feerate when
```
"User Darius" is ambiguous. Maybe "User *Darius* ..." or "A user, Darius, ..."

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+
+- [Bitcoin Core #19550][] adds a new `getindexinfo` RPC that lists each
+  optional index that has been enabled, how many blocks have been
+  indexed so far, and whether those blocks constitute syncing all the way
```
maybe "whether those blocks are synced to the tip of the block chain" (simpler)

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+  implementations.
+
+- [BOLTS #785][] updates the minimum CLTV expiry delta to 18 blocks.  To
+  ensure that the latest state gets recorded onchain, a channel should
```
s/gets/is/

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+
+- [BOLTS #785][] updates the minimum CLTV expiry delta to 18 blocks.  To
+  ensure that the latest state gets recorded onchain, a channel should
+  be unilaterally closed when there are only this many blocks until an
```
this = minimum expiry delta?

jonatack commented on _posts/en/newsletters/2020-08-26-newsletter.md:
```diff
@@ -0,0 +1,164 @@
+  routes with low CLTV expiry deltas, which has in turn led some users
+  to set their deltas to values that are especially unsafe.  This new
+  minimum should help prevent inexperienced users from naively setting
+  an unsafe value.
```
"unsafe" mentioned twice without saying why it is unsafe. A risk associated with higher expiry values is described above; the same for low values may be good here

harding replied:
I think it is explained, "To ensure that the latest state is recorded onchain [before] an LN payment has to be settled."

jonatack reviewed (commented):
One last read-through from me; feel free to ignore if too late

bitschmidty commented:
ACK 446486b

Pushed small style commit.

bitschmidty commented:
Rebased and merged!

@harding @dongcarl @jonatack writing segments this week 👍 
@jonatack @adamjonas @jnewbery 🔎 ✔️ 🙏
